packages/protocol-http/src/headersProxy.spec.ts

```diff
@@ -0,0 +1,156 @@
+import { Fields } from "./Fields";
+import { getHeadersProxy, headersToFields, headerValueToFieldValues } from "./headersProxy";
+
+describe("getHeadersProxy", () => {
+  const mockHeaders = {
+    foo: "bar",
+    baz: "qux,quux",
+  };
+  const mockFields = Fields.from([
+    { name: "foo", values: ["bar"] },
+    { name: "baz", values: ["qux", "quux"] },
+  ]);
+
+  describe("proxy works like a normal Record", () => {
+    describe("access and mutation", () => {
+      it("can get and set individual keys", () => {
+        const headers = getHeadersProxy(new Fields({}));
+        headers["foo"] = "bar";
+        headers.baz = "qux,quux";
```

Comment:
What if commas come with spaces?

Reply:
Whitespace is preserved. I have a test for it [here](https://github.com/aws/aws-sdk-js-v3/pull/4514/files#diff-6d4dc8b75cf95d8327296e6bc2c7b4cd3166d09b8215b3972990d6192728b1edR152-R155), but do you think its worth having one in the getHeadersProxy tests?

Reply:
it's ok if you can't think of any problems it would cause